spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/hazelcast/HazelcastAutoConfigurationTests.java

```diff
@@ -39,6 +39,8 @@
 	@Test
 	public void defaultConfigFile() {
 		// no hazelcast-client.xml and hazelcast.xml is present in root classpath
+		// this also asserts that XML has priority over YAML
```

Comment:
It seems wrong to me that XML on the classpath has priority over YAML on the file system. Is this really what Hazelcast does?

Reply:
yeah, as explained in the other comment.